src/server/scriptVersionCache.ts

```diff
@@ -582,7 +582,8 @@ export class LineIndex {
     }
 }
 
-/** @internal */
+// TODO: knip doesn't see that LineNode is visible via LineIndexWalker

```

Comment:
These are what caused the other baseline change; these types are exported and referenced from other declarations in this file, but are being treated as removable when they shouldn't be.

Reply:
Then I think we should `@knipignore` it. From Knip's perspective, it's an export not used outside this file.

Reply:
Will knip fix this or does this need to be knip ignore. 
EIther way there shouldnt be TODO

Reply:
I would too prefer if knip detected this transitive use, but I don't know if it's a hard blocker either.

Reply:
The `LineNode` class is not used outside of this file, so it would be reported as an unused export.

The [`ignoreExportsUsedInFile` config in `knip.jsonc`](https://github.com/microsoft/TypeScript/pull/56817/files#diff-48d5ba4681726b42e98dae10c08bd94f7f9836644c8f183c6475d10dcf67ebf1R32-R36) prevents unused exported types from being reported, but not values (or classes). I'd say this configuration is good, and the `@knipignore` exception is fine here.

Reply:
But i dont understand why? exported interface `LineIndexWalker` uses it so `LineNode` also need to be exported?

Reply:
Knip treats values and types the same in this regard. Values referenced in something that's exported in the same module don't need to be exported themselves.

Reply:
Which is definitely not correct though. Types need to be exported if that type is exported from the module otherwise it generates declaration errors. 

Anyways i am not saying this needs to be fixed before we merge this but this definitely seems like issue with knip. 
Would prefer not to have TODO here though if knip isnt going to take action.

Reply:
Agreed, described current behavior. Opened an issue for this: https://github.com/webpro-nl/knip/issues/687

Reply:
Since not merged yet, I took the liberty to update this PR with a patched Knip. Also, it tells you when tags are obsolete: https://github.com/microsoft/TypeScript/actions/runs/9662009659/job/26650991993#step:5:16

This is just a hint for now (i.e. does not alter exit code), in a next major it might become an error.

Reply:
Lost this thread (so many collapsed conversations), but I think the above change was overly conservative :( https://github.com/microsoft/TypeScript/pull/56817#issuecomment-2189462434

Reply:
Does this mean we can remove `knipignore` here?

Reply:
Yes, they're now gone.